### ISSUE TITLE
[rl] PPG aux phase recomputes shared trunk twice per minibatch (no fused forward)

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#510]
- **Files:** `crates/rlevo-reinforcement-learning/src/algorithms/ppg/ppg_policy.rs`, `crates/rlevo-reinforcement-learning/src/algorithms/ppg/policies/categorical.rs`, `crates/rlevo-reinforcement-learning/src/algorithms/ppg/ppg_agent.rs`
- **Body:** Confirmed current (2026-07-20). In `ppg_agent.rs`'s aux-phase minibatch loop:

```rust
let aux_v_pred = PpgAuxValueHead::aux_value(self.policy(), obs_t.clone()); // ppg_agent.rs:748
let new_logits = PpgAuxValueHead::logits(self.policy(), obs_t);           // ppg_agent.rs:750
```

`PpgCategoricalPolicyHead::aux_value` (`categorical.rs:108-111`) and `::logits` (`categorical.rs:113-116`) each independently call `self.trunk(obs)` (`fc1` → `tanh` → `fc2` → `tanh`, `categorical.rs:101-104`), so the trunk runs twice on identical input per minibatch, across `e_aux` epochs. The `obs_t.clone()` at the call site exists only because the trait's two methods each take ownership of `obs`.

Fix: add a fused default method on `PpgAuxValueHead`, e.g. `fn aux_value_and_logits(&self, obs: Tensor<B, DB>) -> (Tensor<B, 1>, Tensor<B, 2>)` with a default that calls the two existing methods (so it's additive/non-breaking), and override it on `PpgCategoricalPolicyHead` to share one `trunk(obs)` call. Update `ppg_agent.rs:748-750` to call the fused method and drop the `.clone()`.
- **Severity:** medium (real 2x trunk-FLOPs + one extra clone per aux minibatch across `e_aux` epochs; not "High" since the trunk here is a small 2-layer MLP, not a hot rollout-scale cost)
- **Source:** Code review 2026-06-30
